edge-functions/json-response/middleware.ts

```diff
@@ -0,0 +1,16 @@
+/* eslint-disable @next/next/no-server-import-in-page */
+import { NextResponse, NextRequest } from 'next/server'
+
+export const config = {
+  matcher: '/',
+}
+
+export function middleware(req: NextRequest) {
+  if (req.nextUrl.pathname == '/file.json') {
+    return
+  } else {
+    const url = req.nextUrl
+    url.pathname = '/file.json'
+    return NextResponse.rewrite(url)
```

Comment:
Lets use an API route instead of a public file, soon after releasing middleware we'll release a way for API routes to be on Middleware and in that case this middleware will be removed in favor of the API route.

Reply:
Great feedback, thanks! done